### PR TITLE
[Cairo 1] Add string hash

### PR DESCRIPTION
**Pull request by piwonskp**
I left `wm_string_hash` commented as I'm not sure if it's needed anymore

### CODE REVIEW

File: warplib/string_hash.cairo [hunk 1]
```diff
@@ -1,33 +1,40 @@
...
 }
 
+// func wm_string_hash{pedersen_ptr: HashBuiltin*, range_check_ptr, warp_memory: DictAccess*}(
+//     mem_loc: felt
+// ) -> felt {
+//     let (len, ptr) = wm_to_felt_array(mem_loc);
+
+//     let hashedValue = string_hash(len, ptr);
+//     return hashedValue;
+// }
```
Comment (rodrodros) [inline warplib/string_hash.cairo:33]: Normal `string_hash` is used when the array is a normal cairo array, but `wm_string_hash` is used when it is scattered  in a dict access where it transforms it into a normal cairo array, and then applies `string_hash`. I wouldn't mind for it to stay commented a little longer in the code

**Review by rodrodros — APPROVED** [review]

File: warplib/string_hash.cairo (hunk 1 above)
Comment (piwonskp) [reply warplib/string_hash.cairo:33]: I see, then let's leave it for now. We'll see what to do with that later

**Review by cicr99 — APPROVED** [review]
LGTM

**Comment by piwonskp** [comment]
Replaced by #994

**Pull request closed without merging**